index.js

```diff
@@ -6,21 +6,38 @@ const host = core.getInput('host');
 const port = core.getInput('port');
 const socketPath = core.getInput('socket-path');
 const key = core.getInput('key');
+const lifetimeInSeconds = core.getInput('lifetime');
 
 console.log(`Attempting to create ${socketPath}...`);
-try {
-    execSync(
-        'mkdir -p ~/.ssh && ' +
-        `ssh-keyscan${port ? ` -p ${port}` : ''} "${host}" >> ~/.ssh/known_hosts && ` +
-        `eval $(ssh-agent -a "${socketPath}") && ` +
-        `echo "${key}" | base64 -d | ssh-add -`
-    );
-} catch (e) {
-    console.error(e.message);
-    process.exit(1);
+
+function execSyncSafelyAndReturn(command) {
+    try {
+        return execSync(command, {
+            encoding: 'utf-8'
+        })
+    } catch (e) {
+        if (e.message.includes('Address already in use')) {
+            core.info('Agent already exists on sock. Skipping creation.');
+        } else {
+            core.setFailed(e.message);
+        }
+    }
 }
 
-console.log(`Created ${socketPath}`);
-core.setOutput('socket-path', socketPath);
+// Prepare the host file.
+execSync('mkdir -p ~/.ssh');
+execSync('touch ~/.ssh/known_hosts');
+execSync(`sed -i -e '/^${host} /d' ~/.ssh/known_hosts`);
+execSync(`ssh-keyscan${port ? ` -p ${port}` : ''} "${host}" >> ~/.ssh/known_hosts`);
 
-console.log('Done; exiting.');
+// Start the agent (or re-use one)
+execSyncSafelyAndReturn(`ssh-agent -a "${socketPath}"`);
+const pid = execSyncSafelyAndReturn(`lsof -Fp ${socketPath} | head -n 1 | sed 's/^p//'`)
```

Comment:
You might move the parseInt to be earlier up here

Reply:
done.